pkg/client/client.go

```diff
@@ -11,15 +11,22 @@ import (
 	"fmt"
 	"net/url"
 	"os"
+	"time"
 
 	"github.com/Azure/aks-tls-bootstrap-client/pkg/datamodel"
 	secureTLSBootstrapService "github.com/Azure/aks-tls-bootstrap-client/pkg/protos"
 	"go.uber.org/zap"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/transport"
+	certutil "k8s.io/client-go/util/cert"
 )
 
 // TLSBootstrapClient retrieves tokens for performing node TLS bootstrapping.
 type TLSBootstrapClient interface {
 	GetBootstrapToken(ctx context.Context) (string, error)
+	isKubeConfigStillValid() (bool, error)
```

Comment:
would rather not add this to the interface, though I think we can get rid of this interface altogether - something to revisit (sooner rather than) later